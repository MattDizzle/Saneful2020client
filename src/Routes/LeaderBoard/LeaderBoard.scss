.LeaderBoard{
  width: 100vw;
}

.LeaderBoard h2{
  font-size: 1rem;
  margin: 2.5rem;
}

.leader-list{
  max-width: 520px;
  margin: 0 auto;
  padding: 0;
}

.leader-list li{
  display: flex;
  flex-direction: row;
  padding: 0 0 0 4%;
}

.color-rotate{
  background-color: rgba(68, 91, 192, 0.274);
}

.player-info{
  text-align: left;
}

.player-info h3{
  font-size: 0.95rem;
}

.player-info p{
  font-size: 0.72rem;
  color: darkgray;
  padding-left: 42px;
}
