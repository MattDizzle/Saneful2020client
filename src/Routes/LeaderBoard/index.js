import React, { useEffect, useState } from 'react';
import convertTime from '../../Utils/convert-time';
import SaveApiService from '../../services/save-service';

import './LeaderBoard.scss';

const LeaderBoard = () => {

  const [list, setList] = useState([]);

  useEffect(() => {
    SaveApiService.getLeaderboard()
      .then(res => {
        setList(res);
      })
      .catch((error) => {
        console.error(error);
      });
  }, []);

  const renderList = () => {
    let x = 0;
    let options = ['', 'color-rotate'];

    return list.map((player, index) => {
      let convertedTime = convertTime(player.elapsed_time);

      x = x === 0 ? 1 : 0;

      return (
        <li key={index} className={options[x]}>
          <div className='player-info'>
            <h3>{`${index + 1}. `}{player.user_name}</h3>
            <p>{`${convertedTime.days} day(s), ${convertedTime.hours} hours ${convertedTime.mins} mins`}  {`(${player.dead ? 'deceased' : 'alive'})`}</p>
          </div>
        </li>
      );
    });
  };

  return (
    <section className='LeaderBoard'>
      <h2>Those who made it the longest...</h2>
      <ol className='leader-list'>
        {renderList()}
      </ol>
    </section>
  );
};

export default LeaderBoard;