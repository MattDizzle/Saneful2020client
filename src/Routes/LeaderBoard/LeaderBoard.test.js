import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import LeaderBoard from './index';


describe(`LeaderBoard Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <LeaderBoard />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <LeaderBoard />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})