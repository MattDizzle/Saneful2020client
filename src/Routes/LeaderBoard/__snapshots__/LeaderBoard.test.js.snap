// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoard Component Snapshot test Renders the UI as expected 1`] = `
<section
  className="LeaderBoard"
>
  <h2>
    Those who made it the longest...
  </h2>
  <ol
    className="leader-list"
  />
</section>
`;
